src/AWS.Deploy.CLI/Commands/TypeHints/ExistingSecurityGroupsCommand.cs

```diff
@@ -37,30 +36,53 @@ private async Task<List<SecurityGroup>> GetData(Recommendation recommendation, O
             return await _awsResourceQueryer.DescribeSecurityGroups(vpcId);
         }
 
-        public async Task<List<TypeHintResource>?> GetResources(Recommendation recommendation, OptionSettingItem optionSetting)
+        public async Task<TypeHintResourceTable> GetResources(Recommendation recommendation, OptionSettingItem optionSetting)
         {
             var securityGroups = await GetData(recommendation, optionSetting);
-            return securityGroups.Select(securityGroup => new TypeHintResource(securityGroup.GroupId, securityGroup.GroupId)).ToList();
+
+            var resourceTable = new TypeHintResourceTable
+            {
+                Columns = new List<TypeHintResourceColumn>()
+                {
+                    new TypeHintResourceColumn("groupName", "Group Name"),
+                    new TypeHintResourceColumn("groupId", "Group Id"),
+                    new TypeHintResourceColumn("vpcId", "VPC Id")
+                }
+            };
+
+            foreach (var securityGroup in securityGroups.OrderBy(securityGroup => securityGroup.VpcId))
+            {
+                var row = new TypeHintResource(securityGroup.GroupId, securityGroup.GroupId);
+                row.ColumnValues["groupName"] = securityGroup.GroupName;
+                row.ColumnValues["groupId"] = securityGroup.GroupId;
+                row.ColumnValues["vpcId"] = securityGroup.VpcId;
+
+                resourceTable.Rows.Add(row);
+            }
+
+            return resourceTable;
         }
 
         public async Task<object> Execute(Recommendation recommendation, OptionSettingItem optionSetting)
         {
-            var availableSecurityGroups = (await GetData(recommendation, optionSetting)).OrderBy(x => x.VpcId).ToList();
+            var resourceTable = await GetResources(recommendation, optionSetting);
```

Comment:
I prefer if we use GetData and not GetResources, since we'd be doing extra work without a practical benefit. We'd need to loop over the security groups to create the resource table and then loop over the resource table to display the values.

Reply:
I intentionally routed it through `GetResources`, in my opinion we shouldn't have two different code paths for _"prepare these resources for display"_ between CLI and server mode, that's partially what led to this task in the first place. I was tempted to rip out `GetData` entirely.

It's still one network call, I don't think we're at the point of optimizing the looping yet.